etc/docker-scripts/bootstrapper.py

```diff
@@ -18,243 +18,334 @@
+    """Perform Notebook File Operation"""
+
+    def execute(self) -> None:
+        """Execute the Notebook and upload results to object storage"""
+        notebook = os.path.basename(self.filepath)
+        notebook_name = notebook.replace('.ipynb', '')
+        notebook_output = notebook_name + '-output.ipynb'
+        notebook_html = notebook_name + '.html'
+
+        print("Executing notebook through Papermill: {} ==> {}".format(notebook, notebook_output))
+        try:
+            import papermill
+            papermill.execute_notebook(
+                notebook,
+                notebook_output,
+                kernel_name="python3"
+                # parameters=
+            )
+
+            NotebookFileOp.convert_notebook_to_html(notebook_output, notebook_html)
+            print("Uploading result Notebook back to Object Storage")
+            self.put_file_to_object_storage(notebook_output, notebook)
+            self.put_file_to_object_storage(notebook_html)
+
+            self.process_outputs()
+        except Exception as ex:
+            # log in case of errors
+            print("Unexpected error:", sys.exc_info()[0])
+            NotebookFileOp.convert_notebook_to_html(notebook_output, notebook_html)
+
+            print("Uploading errored Notebook back to Object Storage")
+            self.put_file_to_object_storage(notebook_output, notebook)
+            self.put_file_to_object_storage(notebook_html)
+            raise ex
+
+    @staticmethod
+    def convert_notebook_to_html(notebook_file: str, html_file: str) -> str:
+        """
+        Function to convert a Jupyter notebook file (.ipynb) into an html file
+
+        :param notebook_file: object storage client
+        :param html_file: name of what the html output file should be
+        :return: html_file: the converted notebook in html format
+        """
+        import nbconvert
+        import nbformat
+
+        print("Converting from ipynb to html....")
+        nb = nbformat.read(notebook_file, as_version=4)
+        html_exporter = nbconvert.HTMLExporter()
+        data, resources = html_exporter.from_notebook_node(nb)
+        with open(html_file, "w") as f:
+            f.write(data)
+            f.close()
+
+        return html_file
+
+
+class PythonFileOp(FileOpBase):
+    """Perform Python File Operation"""
+
+    def execute(self) -> None:
+        """Execute the Python script and upload results to object storage"""
+        python_script = os.path.basename(self.filepath)
+        python_script_name = python_script.replace('.py', '')
+        python_script_output = python_script_name + '.log'
+
+        print("Executing Python Script : {} ==> {}".format(python_script, python_script_output))
+        try:
+            # with open(python_script_output, "w") as output_file:
+            #     subprocess.check_call([sys.executable, python_script], stdout=output_file)
+
+            subprocess.check_call(['python', python_script])
```

Comment:
We should probably use [`subprocess.run()`](https://docs.python.org/3/library/subprocess.html#subprocess.run) since that's recommended over the older `check_call()` approach - especially since we only support 3.5+.